inference/core/registries/roboflow.py

```diff
@@ -80,26 +85,38 @@ def get_model_type(model_id: str, api_key: str) -> Tuple[TaskType, ModelType]:
     )
     if cached_metadata is not None:
         return cached_metadata[0], cached_metadata[1]
-    workspace_id = get_roboflow_workspace(api_key=api_key)
-    project_task_type = get_roboflow_dataset_type(
-        api_key=api_key, workspace_id=workspace_id, dataset_id=dataset_id
-    )
     if version_id == STUB_VERSION_ID:
+        if api_key is None:
+            raise MissingApiKeyError(
+                "Stub model version provided but no API key was provided. API key is required to load stub models."
+            )
+        workspace_id = get_roboflow_workspace(api_key=api_key)
+        project_task_type = get_roboflow_dataset_type(
+            api_key=api_key, workspace_id=workspace_id, dataset_id=dataset_id
+        )
         model_type = "stub"
-    else:
-        model_type = get_roboflow_model_type(
-            api_key=api_key,
-            workspace_id=workspace_id,
+        save_model_metadata_in_cache(
             dataset_id=dataset_id,
             version_id=version_id,
             project_task_type=project_task_type,
+            model_type=model_type,
         )
+        return project_task_type, model_type
+    api_data = get_roboflow_model_data(
```

Comment:
I have suspicions that in some cases this piece of code may be faulty.

In case of malfromed response, when `ort` key is not present - we will get `api_data = None` which will raise exception in the next line.

Raised error will not have a meaningful message. At the same time, we have `ModelArtefactError` which we use for that cases